java/src/main/java/org/wildfly/openssl/Identification.java

```diff
@@ -47,7 +50,13 @@
                     } else {
                         sysOs = sysOs.toUpperCase(Locale.US);
                         if (sysOs.startsWith("LINUX")) {
-                            osName = "linux";
+                            String sysVersion = System.getProperty("os.version");
+                            Matcher m = RHEL_PATTERN.matcher(sysVersion);
+                            if (m.matches()) {
+                                osName = m.group(1); // el6, el7, or el8
```

Comment:
One question - since there are different natives attached for WildFly and JBoss EAP (different paths in distro), I am a bit afraid that if I take WildFly and start it on RHEL6,7,8 now, it won't find correct natives automatically, as it will try to search path that is present only in JBoss EAP distro. Is it correct?

Reply:
@jstourac Unfortunately, yes. For community, we only provide `linux-x86_64` and `linux-i386` versions. So that means that with this change, if a user wants to start WildFly on RHEL 6, RHEL 7, or RHEL 8, either the `-Djboss.modules.os-name=linux` property would need to be specified to attempt to make use of what we provide or the user would need to build the appropriate `el6-{i386,x86_64}`, `el7-x86_64`, and `el8_x86_64` module first. However, since making use of the `linux-x86_64` or `linux-i386` natives could potentially result in issues on RHEL (as described in [JBEAP-18856](https://issues.redhat.com/browse/JBEAP-18856), we might not want to default to the generic `linux-x86_64` or `linux-i386` versions in this case anyway. What do you think?

Reply:
@fjuma well, yes, I understand your point. Although, if we don't want to rely on them, why do we ship them at all then :)

My point here is that this will lead in inconsistent behavior of WildFly on different platforms. Making it less usable on RHELs where users will need to define the property, while on other platforms not.

Reply:
@fjuma I wonder if we should add `linux-$cpu-name` by default if `el*` is found. Then it would be on the search path and fallback to that if the el specific one is not found.

Reply:
+1, I've updated this so that the `linux-$cpu-name` variants will get added to the native search path when running on RHEL. This will ensure that we fallback to the ``linux-$cpu-name` modules if the `el` specific modules aren't found.

Reply:
You beat me to it :) https://github.com/wildfly-security/wildfly-openssl/commit/6a99e69cf3f5fe393a558a826a32812abfc0113f

Reply:
:)

Reply:
Thank you, @fjuma and @jamezp!